pyteal/ast/tmpl.py

```diff
@@ -5,15 +5,47 @@
 class Tmpl(LeafExpr):
     """Template expression for creating placeholder values."""
 
-    def __init__(self, tmpl_v: str) -> None:
-        valid_tmpl(tmpl_v)
-        self.name = tmpl_v
+    def __init__(self, op: str, type: TealType, name: str) -> None:
+        valid_tmpl(name)
+        self.op = op
+        self.type = type
+        self.name = name
 
     def __str__(self):
-        return self.name
+        return "(Tmpl {} {})".format(self.op, self.name)
 
     def __teal__(self):
-        raise TealInternalError("Tmpl is not expected here")
+        return [[self.op, self.name]]
 
     def type_of(self):
-        raise TealInternalError("Tmpl is not expected here")
+        return self.type
+    
+    @classmethod
+    def Int(cls, placeholder: str):
+        """Create a new Int template.
+
+        Args:
+            placeholder: The name to use for this template variable. Must start with "TMPL_" and
+            only consist of uppercase alphanumeric characters and underscores.
+        """
+        return cls("int", TealType.uint64, placeholder)
+    
+    @classmethod
+    def Bytes(cls, placeholder: str):
+        """Create a new Bytes template.
+
+        Args:
+            placeholder: The name to use for this template variable. Must start with "TMPL_" and
+            only consist of uppercase alphanumeric characters and underscores.
+        """
+        return cls("byte", TealType.bytes, placeholder)
+
+    @classmethod
+    def Addr(cls, placeholder: str):
```

Comment:
Is the difference between `Bytes` and `Addr` here purely cosmetic?

Reply:
Not entirely. Since `addr` takes in a base32 Algorand address with checksum, you would have to discard the checksum to import it as a normal byte string.